src/server/ProcessList.cpp

```diff
@@ -30,51 +27,29 @@ using namespace Microsoft::Console::Interactivity;
 [[nodiscard]] HRESULT ConsoleProcessList::AllocProcessData(const DWORD dwProcessId,
                                                            const DWORD dwThreadId,
                                                            const ULONG ulProcessGroupId,
-                                                           _In_opt_ ConsoleProcessHandle* const pParentProcessData,
                                                            _Outptr_opt_ ConsoleProcessHandle** const ppProcessData)
 {
-    FAIL_FAST_IF(!(ServiceLocator::LocateGlobals().getConsoleInformation().IsConsoleLocked()));
+    assert(ServiceLocator::LocateGlobals().getConsoleInformation().IsConsoleLocked());
 
     auto pProcessData = FindProcessInList(dwProcessId);
-    if (nullptr != pProcessData)
+    if (pProcessData)
     {
-        // In the GenerateConsoleCtrlEvent it's OK for this process to already have a ProcessData object. However, the other case is someone
-        // connecting to our LPC port and they should only do that once, so we fail subsequent connection attempts.
-        if (nullptr == pParentProcessData)
-        {
-            return E_FAIL;
-            // TODO: MSFT: 9574803 - This fires all the time. Did it always do that?
-            //RETURN_HR(E_FAIL);
-        }
-        else
-        {
-            if (nullptr != ppProcessData)
-            {
-                *ppProcessData = pProcessData;
-            }
-            RETURN_HR(S_OK);
-        }
+        return E_FAIL;
     }
 
     try
     {
-        pProcessData = new ConsoleProcessHandle(dwProcessId,
-                                                dwThreadId,
-                                                ulProcessGroupId);
-
-        // Some applications, when reading the process list through the GetConsoleProcessList API, are expecting
-        // the returned list of attached process IDs to be from newest to oldest.
-        // As such, we have to put the newest process into the head of the list.
-        _processes.push_front(pProcessData);
-
-        if (nullptr != ppProcessData)
-        {
-            *ppProcessData = pProcessData;
-        }
+        pProcessData = new ConsoleProcessHandle{ dwProcessId, dwThreadId, ulProcessGroupId };
+        _processes.emplace_back(pProcessData);

```

Comment:
nit: if this throws, we leak one ConsoleProcessHandle.

We should capture the return value from `emplace_back` and use that to populate `ppProcessData`

Reply:
`_processes` is a list of raw pointers actually unfortunately. I'm going to use `std::unique_ptr` in the meantime until we can refactor this code further to not use raw pointers anymore.

Reply:
Honestly I'm okay with it -- the leak occurs if the emplace fails: `_processes` doesn't take ownership, and we leak it. `unique_ptr` will fix, but just wanted to be clear I'm not as bothered by the list of raw pointers with manual management (though it does gross me out :P)